include/diaspora.php

```diff
@@ -701,6 +701,33 @@ public static function handle_from_contact($contact_id, $gcontact_id = 0) {
 	}
 
 	/**
+	 * @brief get a url (scheme://domain.tld/u/user) from a given Diaspora
+	 * fcontact id or guid
+	 *
+	 * @param mixed $contact_id Either the numeric id or the string guid
+	 *
+	 * @return string the url
+	 */
+	public static function url_from_fcontact($contact_id) {
+		$handle = False;
+
+		logger("fcontact id is ".$contact_id, LOGGER_DEBUG);
+
+		if (is_numeric($contact_id)) {
+			$r = q("SELECT `url` FROM `fcontact` WHERE `id` = %d AND `network` = 'dspr' `url` != ''",
+				intval($contact_id));
+		}else {
+			$r = q("SELECT `url` FROM `fcontact` WHERE `guid` = '%s' AND `network` = 'dspr' AND `url` != ''",
+				$contact_id);
+		}
+		if ($r) {
```

Comment:
Yes, definitely, I got the inspiration from another function in the same file performing a similar lookup, but it would benefit from the change too.